test/integration/integration_test.go

```diff
@@ -169,10 +170,10 @@ func FeatureContext(s *godog.Suite) {
 		getRoutingUrlAndVerifyHTTPResponse)
 
 	// steps for verifying config file content
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
-		configContains)
-	s.Step(`JSON config file "([^"]*)" (contains|does not contain) key "(.*)"(.*)$`,
-		configContains)
+	s.Step(`^JSON config file "([^"]*)" (contains|does not contain) key "(.*)" with value "(.*)"$`,
+		jsonConfigContains)
+	s.Step(`^JSON config file "([^"]*)" (has|does not have) key "(.*)" with value assigned$`,
```

Comment:
@LalatenduMohanty there is an issue with godog, when a line partially matches, and is specified before the other cases, it will take precedence. This way, this can be prevented